File: Data/CodeExamples/HeaderFooterActions.cs
using System;
using System.IO;
using DevExpress.Spreadsheet;

namespace SpreadsheetExamples {
    public static class HeaderFooterActions {
        static void AddHeaderFooter(IWorkbook workbook) {
            #region #AddHeaderFooter
            Worksheet worksheet = workbook.Worksheets["HeaderFooter"];
            workbook.Worksheets.ActiveWorksheet = worksheet;
            WorksheetHeaderFooterOptions headerFooter = worksheet.HeaderFooterOptions;

            // Add headers to first page
            headerFooter.DifferentFirst = true;
            headerFooter.FirstHeader.Left = "File path: " + HeaderFooterCode.WorkbookFilePath + HeaderFooterCode.WorkbookFileName + ".xlsx";
            headerFooter.FirstHeader.Right = "Total number of pages: " + HeaderFooterCode.PageTotal;

            // Add footers to first page, using FromLCR method
            string leftFooter = "Current date: " + HeaderFooterCode.Date;
            string centerFooter = "Current time: " + HeaderFooterCode.Time;
            string rightFooter = "First page";
            headerFooter.FirstFooter.FromLCR(leftFooter, centerFooter, rightFooter);

            // Add header to even pages
            headerFooter.DifferentOddEven = true;
            headerFooter.EvenHeader.Right = "This page number is even: " + HeaderFooterCode.PageNumber;

            // Add footer to odd pages, using FromString method
            string oddPageFooter = HeaderFooterCode.RightSection + "This page number is odd: " + HeaderFooterCode.PageNumber;
            headerFooter.OddFooter.FromString(oddPageFooter);
            #endregion #AddHeaderFooter
        }

        static void AddPicture(IWorkbook workbook, string rootPath) {
            #region #AddPicture
            Worksheet worksheet = workbook.Worksheets["HeaderFooter"];
            workbook.Worksheets.ActiveWorksheet = worksheet;
            WorksheetHeaderFooter oddHeaderFooter = worksheet.HeaderFooterOptions.OddHeader;
            
            // Add a picture to center header
            string filePath = Path.Combine(rootPath + "\\DevExpress.png");
            oddHeaderFooter.AddPicture(filePath, HeaderFooterSection.Center);

            // Change width to fit picture
            oddHeaderFooter.CenterPicture.Width = 500;
            #endregion #AddPicture
        }

        static void RemovePicture(IWorkbook workbook, string rootPath) {
            #region #RemovePicture
            Worksheet worksheet = workbook.Worksheets["HeaderFooter"];
            workbook.Worksheets.ActiveWorksheet = worksheet;
            WorksheetHeaderFooter oddHeaderFooter = worksheet.HeaderFooterOptions.OddHeader;

            // Add a picture to center header
            string filePath = Path.Combine(rootPath + "\\DevExpress.png");
            oddHeaderFooter.AddPicture(filePath, HeaderFooterSection.Center);
            oddHeaderFooter.CenterPicture.Width = 500;

            // Remove picture from center header
            oddHeaderFooter.RemovePicture(HeaderFooterSection.Center);
            #endregion #RemovePicture
        }

        static void FormatPicture(IWorkbook workbook, string rootPath) {
            #region #FormatPicture
            Worksheet worksheet = workbook.Worksheets["HeaderFooter"];
            workbook.Worksheets.ActiveWorksheet = worksheet;
            WorksheetHeaderFooter oddHeaderFooter = worksheet.HeaderFooterOptions.OddHeader;

            // Add a picture to center header
            string filePath = Path.Combine(rootPath + "\\DevExpress.png");
            HeaderFooterPicture picture = oddHeaderFooter.AddPicture(filePath, HeaderFooterSection.Center);

            // Change sizes
            picture.LockAspectRatio = false;
            picture.Width = 500;
            picture.Height = 80;

            // Apply crop
            picture.CropLeft = 10;
            picture.CropRight = 2100;
            picture.CropTop = 10;
            picture.CropBottom = 50;
            #endregion #FormatPicture
        }
    }
}

File: Data/CodeExamples/HeaderFooterActions.vb
Imports System
Imports System.IO
Imports DevExpress.Spreadsheet

Namespace SpreadsheetExamples
	Public NotInheritable Class HeaderFooterActions
		Private Sub New()
		End Sub
		Private Shared Sub AddHeaderFooter(ByVal workbook As IWorkbook)
'			#Region "#AddHeaderFooter"
			Dim worksheet As Worksheet = workbook.Worksheets("HeaderFooter")
			workbook.Worksheets.ActiveWorksheet = worksheet
			Dim headerFooter As WorksheetHeaderFooterOptions = worksheet.HeaderFooterOptions

			' Add headers to first page
			headerFooter.DifferentFirst = True
			headerFooter.FirstHeader.Left = "File path: " & HeaderFooterCode.WorkbookFilePath + HeaderFooterCode.WorkbookFileName & ".xlsx"
			headerFooter.FirstHeader.Right = "Total number of pages: " & HeaderFooterCode.PageTotal

			' Add footers to first page, using FromLCR method
			Dim leftFooter As String = "Current date: " & HeaderFooterCode.Date
			Dim centerFooter As String = "Current time: " & HeaderFooterCode.Time
			Dim rightFooter As String = "First page"
			headerFooter.FirstFooter.FromLCR(leftFooter, centerFooter, rightFooter)

			' Add header to even pages
			headerFooter.DifferentOddEven = True
			headerFooter.EvenHeader.Right = "This page number is even: " & HeaderFooterCode.PageNumber

			' Add footer to odd pages, using FromString method
			Dim oddPageFooter As String = HeaderFooterCode.RightSection & "This page number is odd: " & HeaderFooterCode.PageNumber
			headerFooter.OddFooter.FromString(oddPageFooter)
'			#End Region ' #AddHeaderFooter
		End Sub

		Private Shared Sub AddPicture(ByVal workbook As IWorkbook, ByVal rootPath As String)
'			#Region "#AddPicture"
			Dim worksheet As Worksheet = workbook.Worksheets("HeaderFooter")
			workbook.Worksheets.ActiveWorksheet = worksheet
			Dim oddHeaderFooter As WorksheetHeaderFooter = worksheet.HeaderFooterOptions.OddHeader

			' Add a picture to center header
			Dim filePath As String = Path.Combine(rootPath & "\DevExpress.png")
			oddHeaderFooter.AddPicture(filePath, HeaderFooterSection.Center)

			' Change width to fit picture
			oddHeaderFooter.CenterPicture.Width = 500
'			#End Region ' #AddPicture
		End Sub

		Private Shared Sub RemovePicture(ByVal workbook As IWorkbook, ByVal rootPath As String)
'			#Region "#RemovePicture"
			Dim worksheet As Worksheet = workbook.Worksheets("HeaderFooter")
			workbook.Worksheets.ActiveWorksheet = worksheet
			Dim oddHeaderFooter As WorksheetHeaderFooter = worksheet.HeaderFooterOptions.OddHeader

			' Add a picture to center header
			Dim filePath As String = Path.Combine(rootPath & "\DevExpress.png")
			oddHeaderFooter.AddPicture(filePath, HeaderFooterSection.Center)
			oddHeaderFooter.CenterPicture.Width = 500

			' Remove picture from center header
			oddHeaderFooter.RemovePicture(HeaderFooterSection.Center)
'			#End Region ' #RemovePicture
		End Sub

		Private Shared Sub FormatPicture(ByVal workbook As IWorkbook, ByVal rootPath As String)
'			#Region "#FormatPicture"
			Dim worksheet As Worksheet = workbook.Worksheets("HeaderFooter")
			workbook.Worksheets.ActiveWorksheet = worksheet
			Dim oddHeaderFooter As WorksheetHeaderFooter = worksheet.HeaderFooterOptions.OddHeader

			' Add a picture to center header
			Dim filePath As String = Path.Combine(rootPath & "\DevExpress.png")
			Dim picture As HeaderFooterPicture = oddHeaderFooter.AddPicture(filePath, HeaderFooterSection.Center)

			' Change sizes
			picture.LockAspectRatio = False
			picture.Width = 500
			picture.Height = 80

			' Apply crop
			picture.CropLeft = 10
			picture.CropRight = 2100
			picture.CropTop = 10
			picture.CropBottom = 50
'			#End Region ' #FormatPicture
		End Sub
	End Class
End Namespace